libsolidity/analysis/ReferencesResolver.h

```diff
@@ -85,6 +85,9 @@ class ReferencesResolver: private ASTConstVisitor, private yul::ASTWalker
 	bool visit(InlineAssembly const& _inlineAssembly) override;
 	bool visit(Return const& _return) override;
 
+	/// Do not handle nor recurse into ImportDirective.
+	bool visit(ImportDirective const&) override { return false; }
```

Comment:
Do you get an error if we don't have this? I hope we do.

Reply:
So I think this is the only one that we might need, because the resolving works differently for imports.
Please check that we get an error for:
```
function f() pure {}
import {f as g} from "emptyfile.sol"; // where this is an empty file
```

Reply:
No we don't. This is more a fail-safe to make sure we won't enter by accident.

NameAndTypeResolver seems to be the only user of ReferenceResolver and it's passing a SourceUnit, so this makes sure Identifier from ImportDirective isn't handled (without this I did not get any error though).

Reply:
Can you please clarify what "no we don't" refers to? Can you add a relevant test?

Reply:
The above code causes an error with and without that visit

```
Error: Declaration "f" not found in "emptyfile.sol" (referenced as "emptyfile.sol").
 --> test.sol:3:1:
  |
3 | import {f as g} from "emptyfile.sol"; // where this is an empty file
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^


```

Reply:
So, for this particular code example, what is happening is that the error I posted is triggered in `performImports` which is happeneing before `resolveNamesAndTypes` is called which internally calls the ReferencesResolver and thus the visit is not happening at all.

I am now checking with a modified example

Reply:
In a modified example where `f` exists as contract in `emptyfile.sol`, it does indeed set the wrong referencedDeclaration

Reply:
And no error is caused either.

Reply:
No, I was wrong. It is setting the correct referencedDeclaration when it does visit it in `ReferencesResolver::visit(Identifier..)`

Reply:
To be clear: `f` is set to the contract found in `emptyfile.sol`

I would continue the analysis now, but it's getting too late now

Reply:
And it is correct because the `ImportDirective` gets the scope of the file it is importing. So can visit as normal here and can remove the `return false;` here.

Reply:
It would probably be wise to add the cases you investigated here as test cases.

Reply:
I am pretty sure we already have tons of tests for that simple case I investigated, I mainly traced it to make sure it doesn't just work by accident